org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/TypeBinding.java

```diff
@@ -1559,7 +1559,9 @@ public boolean needsUncheckedConversion(TypeBinding targetType) {
 	targetType = targetType.leafComponentType();
 	if (!(targetType instanceof ReferenceBinding))
 		return false;
-
+	if (!hasBoundParameterizedTypeInHierarchy(targetType)) {
```

Comment:
Looking at JLS I think it suffices to change the previous check to
```java
if (!(targetType instanceof ParameterizedTypeBinding))
    return false;
```

Reply:
I have done this but I'm not completely sure here because `isBoundParameterizedType` only return false if it is `Type<?>`, so I'm not completely sure this is covered here as well.

But maybe then `targetType.isBoundParameterizedType()` would be enough as an extra check? But then one would wonder why further down in the method it is even checked again for enclosing types?